dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -137,6 +140,19 @@ public KernelBuilder WithConfiguration(KernelConfig config)
         return this;
     }
 
+    /// <summary>
+    /// Use the given default trust service with the kernel to be built.
+    /// Functions directly created through the kernel will use this trust service.
+    /// If null, the created functions will rely on the DefaultTrustService implementation.
+    /// </summary>
+    /// <param name="defaultTrustService">Default trust service to use.</param>
+    /// <returns>Updated kernel builder including the given service.</returns>
+    public KernelBuilder WithDefaultTrustService(ITrustService? defaultTrustService)
```

Comment:
nit: I'd just call this WithTrustService(..) to differentiate it from THE `DefaultTrustService`.

Similar to the other With* things (WithLogger, WithMemory), it doesn't prevent caller from overriding with a different instance to a specific function call later on; I don't think this one needs the "Default*" name treatment.

Reply:
For sure, makes sense :)

But I guess we would have to rename attribute in the `ISKFunction` and `IKernel` to something like `TrustServiceInstance`, to avoid name conflicts with the `TrustService` (default) class name.